compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3482,9 +3483,11 @@ object Types {
     private var myWidened: Type = _
 
     private def computeAtoms()(using Context): Atoms =
-      if tp1.hasClassSymbol(defn.NothingClass) then tp2.atoms
-      else if tp2.hasClassSymbol(defn.NothingClass) then tp1.atoms
-      else tp1.atoms | tp2.atoms
+      val tp1n = tp1.normalized
```

Comment:
This part looks like a definite improvement; it could go in independently.

Reply:
@mbovel this is the other atoms fix.

I'm not sure actually if it's a `hasClassSymbol` bug.  Should `hasClassSymbol` normalize?  If I remember correctly the question, which comes from tests/pos/mt-redux-norm.perspective.scala, is does `Tuple.Fold[EmptyTuple, Nothing, [x, y] =>> x | y]` `hasClassSymbol(defn.NothingClass)`?